pkgs/development/python-modules/requests-mock/default.nix

```diff
@@ -1,33 +1,52 @@
-{ lib, buildPythonPackage, fetchPypi, python
-, mock
+{ lib
+, buildPythonPackage
+, fetchPypi
+, fixtures
 , purl
+, pytest
+, python
 , requests
+, requests-futures
 , six
-, testrepository
 , testtools
-, pytest
 }:
 
 buildPythonPackage rec {
   pname = "requests-mock";
-  version = "1.10.0";
+  version = "1.11.0";
+  format = "setuptools";
 
   src = fetchPypi {
     inherit pname version;
-    hash = "sha256-WcnDJBmp+xroPsJC2Y6InEW9fXpl1IN1zCQ+wIRBZYs=";
+    hash = "sha256-7xC1crSJpfKOCbcIaXIIxKOyuJ74Cp8BWENA6jV+w8Q=";
   };
 
-  patchPhase = ''
-    sed -i 's@python@${python.interpreter}@' .testr.conf
-  '';
-
   propagatedBuildInputs = [ requests six ];
 
-  nativeCheckInputs = [ mock purl testrepository testtools pytest ];
+  nativeCheckInputs = [
+    fixtures
+    purl
+    pytest
+    requests-futures
+    testtools
+  ];
+
+  checkPhase = ''
+    runHook preCheck
+
+    # Test Testtools
+    ${python.interpreter} -m testtools.run discover
+
+    # Test PyTest
+    pytest tests/pytest
+
+    runHook postCheck
+  '';
```

Comment:
Can we just use pytestCheckHook?

Reply:
Looks like that works, thank you for the suggestion.